app/presenters/content_embed_presenter.rb

```diff
@@ -0,0 +1,47 @@
+module Presenters
+  class ContentEmbedPresenter
+    def initialize(edition)
+      @edition = edition
+    end
+
+    def render_embedded_content(details)
+      return details if details[:body].nil?
+
+      details[:body] = if details[:body].is_a?(Array)
+                         details[:body].map do |content|
+                           {
+                             content_type: content[:content_type],
+                             content: render_embedded_editions(content[:content]),
+                           }
+                         end
+                       else
+                         render_embedded_editions(details[:body])
+                       end
+
+      details
+    end
+
+  private
+
+    def render_embedded_editions(content)
+      embedded_content_references = EmbeddedContentFinderService.new.find_content_references(content)
+      embedded_content_references_by_content_id = embedded_content_references.index_by(&:content_id)
+      embedded_editions = @edition
+        .links
+        .where(link_type: "embed")
+        .map(&:target_documents)
+        .map { |target_documents| target_documents.find_by(locale: @edition.locale) || target_documents.find_by(locale: Edition::DEFAULT_LOCALE) }
+        .map(&:live)
```

Comment:
There's a bit of an `n+1` here I think. I'd expect this to make:

- 1 query to get `links.where(link_type: "embded")`
- n queries to get target documents for each link
- n queries to get the live edition for each target document

This is probably fine for now, as n probably won't be large (as I don't expect documents to have that more than a few embeds).

I think you could do it in one query that gets all the possible locales, and then find the correct locales in ruby.

Something like:

```ruby
edition_ids = @edition
  .links
  .where(link_type: "embed")
  .joins(target_documents: :live)
  .where(target_documents: { locale: locales })
  .select("editions.id")
# Note ^ the use of select rather than pluck here means this is still
# an `ActiveRecord_AssociationRelation`, rather than an array of ids,
# so it won't make a separate SQL query
  
embedded_editions = Edition.where(id: edition_ids)
# Generates:
# SELECT "editions".* FROM "editions"
# WHERE "editions"."id" IN (
#   SELECT "editions"."id" FROM "links"
#   INNER JOIN "documents" "target_documents"
#     ON "target_documents"."content_id" = "links"."target_content_id"
#   INNER JOIN "editions"
#     ON "editions"."content_store" = 'live'
#     AND "editions"."document_id" = "target_documents"."id"
#   WHERE "links"."edition_id" = 1
#   AND "links"."link_type" = 'embed'
#   AND "target_documents"."locale" IN ('cy', 'en')
# )

# Exercise for the reader: we'll have multiple editions where there are multiple locales. Get the best edition for the locales we want.
```

Reply:
Alternatively, you could use the same trick as we do here: https://github.com/alphagov/publishing-api/blob/c3bfff2652016ec4acbd2883dd22786ab7b8ea7d/app/queries/get_edition_ids_with_fallbacks.rb#L14 which is complicated but presumably works...

Could even use `Queries::GetEditionIdsWithFallbacks` directly I guess, but you might end up making one or two unnecessary queries there.

Reply:
I've gone for the approach of using `Queries::GetEditionIdsWithFallbacks` directly, as that seems to be the least complicated and DRY-est way of getting the embedded editions.

I think that approach should also make it easier for us to deal with embedding draft content (should we wish to) in the future.